dtc-lexer.l

```diff
@@ -207,6 +207,12 @@ static void PRINTF(1, 2) lexical_error(const char *fmt, ...);
 			return DT_PATH_REF;
 		}
 
+<*>\&{LABEL}\/{PATHCHAR}*	{	/* label-relative path reference */
+			DPRINT("Ref: %s\n", yytext+1);
+			yylval.labelref = xstrdup(yytext+1);
+			return DT_REL_REF;
+		}
+
```

Comment:
The basic idea seems good to me: this works with the many similarities between dtc labels and OF aliases (alias relative paths were traditionally allowed in OF).  Lexing this separately from the existing references makes me a little bit nervous though.

On the one hand, PATHCHAR is a sufficiently wide class, that I worry that lexing it without some sort of delimiters around it could lead to ambiguities in certain cases.  On the other hand, my general experience is that it's usually best to make the lexer to accept as much as you reasonably can for a particular token type, then filter it down in the parser.  That tends to lead to more useful error behaviour than having the lexer fall back to a different token type which then feeds nonsense into the parser.

At least for a first pass, I'd suggest making label-relative references a variant of absolute path references, so `&{label/then/subdirs}` rather than `&label/then/subdirs`.  I believe you can have the lexer accept both as the same token type, then later disambiguate them depending on whether they start with `/` or not.

Reply:
Fair enough. I've just force-pushed syntax change to `&{label/then/subdirs}`.